app/controllers/api/solutions/submissions_controller.rb

```diff
@@ -15,6 +15,7 @@ def create
       # rescues SubmissionFileTooLargeError exceptions
       return render_error(400, :file_too_large) if files.any? { |file| file[:content].size > 1.megabyte }
 
+      # TODO: Allow rerunning of tests if previous submission was an error / ops error / timeout
```

Comment:
@iHiD I added this a todo because I don't know how we want to handle this.

Reply:
@iHiD The React bit works already. We just need to conditionally raise the `DuplicateSubmissionError`.

Reply:
I can ignore this now, right?

Reply:
@iHiD Oh, I don't think so yet. We need to ignore raising the error if the last submission errored. Currently, the React component works correctly by not disabling the button, but the backend will still return the `DuplicateSubmissionError`.

Reply:
Ah. I see. But it is still a duplicate submission, so that error is correct. What we need is for a new test-run to init off the old submission I guess. I'll have a think about how best to do this.